### PR TITLE
meta-raspberrypi: refactoring for consistency

### PR DESCRIPTION
**Pull request by hhromic**
I did some refactoring work to `meta-raspberrypi` to be more consistent across all recipes and hence easier to maintain/understand the layer. See below.

* standardised all `files` directories to `${PN}` and adapted bbappends according to [official Yocto recommendations](https://www.yoctoproject.org/docs/latest/ref-manual/ref-manual.html#var-FILESEXTRAPATHS)
* use `_git` suffix for recipes pulling from git repositories without a specific version
* use `_%` suffix for bbappends when missing or instead of specific versions
* standardised installation of udev rules into `/etc/udev/rules.d`
* flattened xserver-xf86-config files directory

Affected recipes:

* bcm2835-bootfiles
* formfactor
* rpi-u-boot-scr
* pi-bluetooth
* psplash
* udev-rules-rpi
* udev-rules-udisks-rpi
* pi-blaster
* wiringpi
* vc-graphics-hardfp
* vc-graphics
* xserver-xf86-config
* linux-raspberrypi-dev
* linux-raspberrypi_4.9
* gstreamer1.0-omx
* x264

I **tested all the affected recipes** and they are all building successfully.

### CODE REVIEW

**Comment by betafive** [comment]
You need to break this down into separate patches that can be reviewed. You have some non-functional whitespace cleanup in with some changes which do affect the resulting images (e.g. standardised installation of udev rules into `/etc/udev/rules.d`).

**Comment by hhromic** [comment]
Thanks for the feedback @betafive.
I thought about doing this one by one but then I realised it might create too many PRs.
Or do you mean to separate into multiple commits in the same PR?
Github allows to comment every part (renames, code change, etc.) of the PR/commit I submitted too.

I am curious about the issue you mention (the non-functional whitespaces). I guess you mean this piece?:
```
SRC_URI = " file://99-com.rules"
```
turned into
```
SRC_URI = "file://99-com.rules"
```

I did build the recipes individually and into an image and they worked fine in my build env using the standard image recipes.
Let me know how should I proceed. Thanks and happy to help.

**Comment by hhromic** [comment]
Ah I see, I got confused. Ok I will separate the cleanups into logical commits in this PR.
Thanks for the patience guys :)

**Comment by hhromic** [comment]
separated refactoring/cleanup into separate commits per recipe now

**Comment by betafive** [comment]
I think I've explained this badly. The idea isn't to split out changes to each recipe into commits, it's to split out functional changes from whitespace changes and cleanups. https://github.com/agherzan/meta-raspberrypi/pull/263/commits/f95536d9369df61c82a366100178f9271b888621 is a good example - this should be at least 3 separate commits (`base_libdir` -> `sysconfdir` change, setting `S` (which I'd argue is unnecessary) and whitespace cleanup).

You can have a single whitespace cleanup patch the touches several recipes, a single `files` -> `${PN}` rename patch which touches several recipes, etc. The idea is to be able to debug more easily later if we find problems. For example, renaming an `_${PV}.bbappend` file to `_%.bbappend' may be something we need to look at again later if the underlying recipe changes but this is logically separate to renaming `${PN}` directories to `files` everywhere.

Does that make sense?

**Comment by hhromic** [comment]
hi @betafive , yes I agree with you.
I re-did the PR now grouping the refactoring, clean-up and functional changes apart from each other.
I hope is clearer now. I'm traveling tomorrow until next week, but I will check the comments once I return.

**Comment by betafive** [comment]
This is where I hate GitHub as I can't comment on commits themselves, so I'll give my review here:

* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/ea53ac55599343c8b9aecbf9f4ff52a98158d0c9: LGTM now
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/1b131735e5cbc4778a47ad956502b17a0be931fe: LGTM
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/644520eb5d2a80daa568aaad8f79b4206fb7fb30: LGTM
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/6ad53e3780394a104eec7b5bdd4c2ec2a2430293: The `rpi` directory is there so that the modifications are only applied to Raspberry Pi devices. I'd say we should keep the current directory structure as-is here.
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/08c9a0315177b84b3d1e38d01f5471112674979b: LGTM
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/2f7aac2f9517cf2b06da3f9a1255f2eca639954c: Not sure what this achieves.
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/530672d981d78876481185d34eecfc9784a9c9b3: LGTM
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/dc8adcc3a0ba7a8025be535306d647c4f735e3df: Why is this change needed?
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/3dc4b7010ac3b40ccc5ca1a8c49b7a7fdd5cfd54: LGTM
* https://github.com/agherzan/meta-raspberrypi/pull/263/commits/3e59ecb7b9f86e87cfea66142f7b6b026141e597: LGTM

One common thing - could you modify the commit summaries so they don't wrap onto a second line?

File: recipes-core/psplash/psplash_git.bbappend [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
```
Comment (kraj) [inline recipes-core/psplash/psplash_git.bbappend]: this should be BPN otherwise it may not work in multilib env.

**Comment by kraj** [comment]
I have questions about this series. So for expediency please divide it into multiple pulls. One could be just the formatting changes and there are more intrusive changes which I think should be dropped.

File: recipes-core/psplash/psplash_git.bbappend (hunk 1 above)
Comment (agherzan) [reply recipes-core/psplash/psplash_git.bbappend]: ++

**Comment by hhromic** [comment]
Hi all, this PR is now just for renaming files and directories (no recipe content changes), with the sole exception of updating how generic licenses from Poky are used. In summary this PR now only contains:

* wiringpi: use lower-case in recipe directory name
* meta-raspberrypi: use generic licenses in ${COMMON_LICENSE_DIR} from Poky
* meta-raspberrypi: use `_git` suffix in recipes without a specific version
* meta-raspberrypi: use `_%` suffix instead of specific versions for bbappends

Those changes are all tested and doesn't change anything beyond cosmetics and files naming.
I will submit more PRs with the other proposed changes separatedly.

@betafive , you asked for commit messages to not be split. This is only happens in GitHub's UI, I can't control it. However, I can assure the messages aren't chopped in the actual git log.

Hopefully all is ok now and can be merged in stages =)

**Comment by hhromic** [comment]
Mmm you are right. I got confused because it fetches from github using a commit hash instead of a particular version tag.
Should I update the recipe name to reflect the date, e.g. `bcm2835-bootfiles-20180417.bb`?
The same then applies for the vc-graphics recipes that also pull from common/firmware.inc.
 I know this implies more work at every update (needing to rename the recipe every time), however it makes clear what version the recipe builds, and is the same approach many recipes in Poky adopt.
My objective is just to make the layer clearer (easy to understand) and manage.

**Comment by hhromic** [comment]
I just noticed that the variables controlling the firmware version/revision are default assignments (`?=`):
```
RPIFW_DATE ?= "20180417"
SRCREV ?= "5db8e4e1c63178e200d6fbea23ed4a9bf4656658"
RPIFW_SRC_URI ?= "https://github.com/raspberrypi/firmware/archive/${SRCREV}.tar.gz"
RPIFW_S ?= "${WORKDIR}/firmware-${SRCREV}"
```
So this makes the firmware version configurable from `local.conf` and renders naming the recipe after a version number such as the date inviable/irrelevant.

Is this an intended feature or should the firmware version be static in the recipe itself like any other package? If the former, then maybe this should be documented and I agree the `_git` suffix doesn't apply.

**Comment by betafive** [comment]
This is intended and it's something I've used a few times. It's undocumented as it's not a "normal" configuration setting (such as RPI_USE_U_BOOT) where we test the different options. So yes, this can be overridden but if you do so you're on your own.

**Comment by hhromic** [comment]
Hi @betafive , thanks for the clarification.
I decided to leave the `bcm2835-bootfiles` and `vc-graphics` recipes untouched for now after your and kraj's remarks, so no `_git` suffix for these. I updated the PR accordingly.

In summary, this PR's changes are:
* meta-raspberrypi: use `_%` suffix instead of specific versions for bbappends
* pi-bluetooth: use `_git` suffix in recipe filename
* meta-raspberrypi: use generic licenses in ${COMMON_LICENSE_DIR} from Poky
* wiringpi: use lower-case in recipe directory name

File: recipes-graphics/vc-graphics/vc-graphics-hardfp_git.bb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 CONFLICTS = "vc-graphics"
 
+FILESEXTRAPATHS_prepend := "${THISDIR}/vc-graphics:"
+
```
[the remark this answers is not in the document]
Comment (hhromic) [reply recipes-graphics/vc-graphics/vc-graphics-hardfp_git.bb]: That change is not in the PR anymore as I decided to study these changes in another PR.
I agree I need to consider all these factors as you mention.

[the remark this answers is not in the document]
Comment (betafive) [reply recipes-graphics/vc-graphics/vc-graphics-hardfp_git.bb]: Not sure why that comment shows up as "32 minutes ago". Stupid GitHub!

This is from previous review and is out-of-date. All looks good now.

**Pull request merged**